code/game/objects/machinery/computer/communications.dm

```diff
@@ -97,6 +97,7 @@
 				to_chat(usr, span_warning("You need to swipe your ID."))
 
 		if("announce")
+			var/mob/living/carbon/human/sender = usr
```

Comment:
declare the variable at the last moment, ie line 120.

You do not declare vars that you may not use as a general rule